interfaces/builtin/nvidia_tegra_drivers_support.go

```diff
@@ -0,0 +1,96 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const nvidiaTegraDriversSupportSummary = `allows hardware access to NVIDIA tegra platforms`
+
+const nvidiaTegraDriversSupportBaseDeclarationSlots = `
+  nvidia-tegra-drivers-support:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const nvidiaTegraDriversSupportConnectedPlugAppArmor = `
+@{PROC}/sys/vm/mmap_min_addr r,
+
+# required to read chip information
+/sys/devices/soc0/platform r,
+/sys/devices/soc0/soc_id r,
+/sys/devices/soc0/revision r,
+/sys/devices/soc0/major r,
+
+# nvidia-smi is trying to access /sys/bus/nvmem/devices/fuse/nvmem
+# but doesn't have permissions to the link location which is listed here
+/sys/devices/platform/bus@0/3810000.fuse/fuse/nvmem r,
+
+/dev/nvmap rw,
+/dev/dri/renderD128 rw,
+/dev/nvgpu/igpu0/power rw,
+/dev/nvgpu/igpu0/ctrl rw,
+/dev/host1x-fence rw,
+
+# tries to create shared memory slab with mknod in IPC cuda apps
+/dev/shm/memmap_ipc_shm rw,
+`
+
+var nvidiaTegraDriversSupportConnectedPlugUdev = []string{
+    // Nvidia dma barrier
+	`SUBSYSTEM=="host1x-fence" OWNER="root" GROUP="video" MODE="0660"`,
+
+    // Tegra memory manager
+	`KERNEL=="nvmap" OWNER="root" GROUP="video" MODE="0660"`,
```

Comment:
```suggestion
	`KERNEL=="nvmap"`,
```

Reply:
The question remains if this is useful for anything the. If the snap using this new support interface is going to actually perform this access or not?

I agree that if we can offload the udev rules to a gadget then this would be much saner and I think, bulk of this might go away?

Reply:
I have no issue with that. I think I initially had difficulties understanding what the udev rules in snapd interfaces are meant to be for but from what I understand now it's simply used to tag devices to be added into the command's cgroup.